lib/Doctrine/ODM/MongoDB/Utility/EncryptedFieldsMapGenerator.php

```diff
@@ -14,30 +14,67 @@
 use function assert;
 use function iterator_to_array;
 
-final class EncryptionFieldMap
+final class EncryptedFieldsMapGenerator
 {
+    private array $generatorStack = [];
+
     public function __construct(private ClassMetadataFactoryInterface $classMetadataFactory)
     {
     }
 
+    /**
+     * Returns the full encryption fields map for a document manager
+     *
+     * @return array<class-string, array<string, array>>
+     */
+    public function getEncryptedFieldsMap(): array
+    {
+        $encryptedFieldsMap = [];
+
+        $allMetadata = $this->classMetadataFactory->getAllMetadata();
+        foreach ($allMetadata as $classMetadata) {
+            if (! $classMetadata->isDocument()) {
+                continue;
+            }
+
+            $classMap = iterator_to_array($this->createEncryptedFieldsMapForClass($classMetadata));
+            if ($classMap === []) {
+                continue;
+            }
+
+            $encryptedFieldsMap[$classMetadata->getName()] = $classMap;
```

Comment:
Since this is a database configuration, it would make more sense to use the collection name as key.

```suggestion
            $encryptedFieldsMap[$classMetadata->getCollection()] = $classMap;
```

Reply:
When dumping the encryptedFieldsMap, we need the full namespace. However, that namespace depends on the `db` property in metadata which may be empty, in which case it falls back to the default database from the configuration. I figured it was best to return the class name as that provides useful information both for printing the map and allows fetching the information necessary to assemble the full namespace.